orion/primitives/tadgan.py

```diff
@@ -247,9 +374,32 @@ def fit(self, X, **kwargs):
             X (ndarray):
                 N-dimensional array containing the input training sequences for the model.
         """
-        self._build_tadgan(**kwargs)
-        X = X.reshape((-1, self.shape[0], 1))
-        self._fit(X)
+        if self.validation_split > 0:
+            valid_length = round(len(X) * self.validation_split)
+            train = X[:-valid_length].copy()
+            valid = X[-valid_length:].copy()
+
+            valid = valid.astype(np.float32)
+            valid = tf.data.Dataset.from_tensor_slices(valid).shuffle(valid.shape[0])
+            valid = valid.batch(self.batch_size, drop_remainder=True)
+
+            callbacks = [
+                callback['class'](**callback.get('args', dict()))
+                for callback in self.callbacks
+            ]
+
+        else:
+            train = X.copy()
+            valid = None
+            callbacks = None
+
+        train = train.astype(np.float32)
```

Comment:
Is there a specific reason for this `32` ? Shouldn't we use `np.float` instead ? This can cause some precision errors.